bundles/org.openhab.binding.vektiva/src/main/java/org/openhab/binding/vektiva/handler/VektivaSmarwiHandler.java

```diff
@@ -0,0 +1,263 @@
+            logger.debug("Received command: {}", command);
+            String cmd = getSmarwiCommand(command);
+            if (COMMAND_OPEN.equals(cmd) || COMMAND_CLOSE.equals(cmd) || COMMAND_STOP.equals(cmd)) {
+                if (RESPONSE_OK.equals(sendCommand(cmd))) {
+                    lastPosition = COMMAND_OPEN.equals(cmd) ? 0 : 100;
+                }
+            }
+            if (command instanceof PercentType) {
+                if (RESPONSE_OK.equals(sendCommand(COMMAND_OPEN + "/" + cmd))) {
+                    lastPosition = Integer.parseInt(cmd);
+                }
+            }
+        }
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        if (future != null && !(future.isCancelled() || future.isDone())) {
+            future.cancel(true);
+        }
+        if (httpClient.isStarted()) {
+            try {
+                httpClient.stop();
+            } catch (Exception e) {
+                //silence
+            }
+        }
+        closeSession();
+        if (webSocketClient.isStarted()) {
+            try {
+                webSocketClient.stop();
+            } catch (Exception e) {
+                //silence
+            }
+        }
+    }
+
+    private void closeSession() {
+        if (session != null && session.isOpen()) {
+            session.close();
+        }
+    }
+
+    private String getSmarwiCommand(Command command) {
+        switch (command.toString()) {
+            case "UP":
+                return COMMAND_OPEN;
+            case "DOWN":
+                return COMMAND_CLOSE;
+            case "STOP":
+                return COMMAND_STOP;
+            default:
+                return command.toString();
+        }
+    }
+
+    private @Nullable String sendCommand(String cmd) {
+        String url = "http://" + config.ip + "/cmd/" + cmd;
+
+        try {
+            ContentResponse resp = httpClient.newRequest(url).method(HttpMethod.GET).send();
+            logger.trace("Response: {}", resp.getContentAsString());
+            if (resp.getStatus() == 200) {
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE);
+            }
+            return resp.getContentAsString();
+        } catch (InterruptedException e) {
+            logger.error("API execution has been interrupted", e);
+            updateStatus(ThingStatus.OFFLINE);
+        } catch (TimeoutException e) {
+            logger.error("Timeout during API execution", e);
+            updateStatus(ThingStatus.OFFLINE);
+        } catch (ExecutionException e) {
+            logger.error("Exception during API execution", e);
+            updateStatus(ThingStatus.OFFLINE);
+        }
+        return null;
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(VektivaSmarwiConfiguration.class);
+        logger.debug("IP address: {}", config.ip);
+
+        try {
+            httpClient.start();
```

Comment:
Really? Tried both HttpClient and WebSockek and both are stopped.
```
 [vektiva.handler.VektivaSmarwiHandler] - Exception during status update
java.util.concurrent.ExecutionException: java.util.concurrent.RejectedExecutionException: org.eclipse.jetty.client.HttpClient@26b1ac02 is stopped
        at org.eclipse.jetty.client.util.FutureResponseListener.getResult(FutureResponseListener.java:118) ~[?:?]
        at org.eclipse.jetty.client.util.FutureResponseListener.get(FutureResponseListener.java:101) ~[?:?]
        at org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:684) ~[?:?]
        at org.openhab.binding.vektiva.handler.VektivaSmarwiHandler.checkStatus(VektivaSmarwiHandler.java:181) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:?]
        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) [?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
        at java.lang.Thread.run(Thread.java:748) [?:?]
Caused by: java.util.concurrent.RejectedExecutionException: org.eclipse.jetty.client.HttpClient@26b1ac02 is stopped
        at org.eclipse.jetty.client.HttpDestination.send(HttpDestination.java:282) ~[?:?]
        at org.eclipse.jetty.client.HttpDestination.send(HttpDestination.java:247) ~[?:?]
        at org.eclipse.jetty.client.HttpClient.send(HttpClient.java:573) ~[?:?]
        at org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:727) ~[?:?]
        at org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:680) ~[?:?]
```
I even checked the state in VektivaSmarwiHandler constructor and they are both stopped...or this is a new feature of OH2.5? I am still using OH2.4...
Thanks

Reply:
I submitted a PR (https://github.com/octa22/openhab2-addons/pull/3) against your branch which works fine in my environment - okay, I did not test the binding itself because I do not own such devices - but both `HttpClient` and `WebSocket` are working.

It resolves every comment I added recently.